build-logic/publishing/src/main/kotlin/build-logic.publish-to-local-repo.gradle.kts

```diff
@@ -0,0 +1,41 @@
+plugins {
+    id("java-library")
+    id("maven-publish")
+}
+
+val localRepoElements by configurations.creating {
+    isCanBeConsumed = true
+    isCanBeResolved = false
+    description =
+        "Shares local maven repository directory that contains the artifacts produced by the current project"
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("maven-repository"))
+        attribute(Bundling.BUNDLING_ATTRIBUTE, objects.named(Bundling.EXTERNAL))
+    }
+}
+
+val localRepoDir = layout.buildDirectory.dir("local-maven-repo")
+
+publishing {
+    repositories {
+        maven {
+            name = "local"
```

Comment:
I know we control the build, but would it make sense for this have a more unique name?

Reply:
Any suggestions?

Reply:
hah, yeah sorry I guess my comment was weakly actionable.

localProjectModules?

Reply:
The repo will include single module only :)
Each "module" creates its own repo under `build/...`, and publishes there.

It might be `tmp-maven-repo` would be better.

Reply:
Sgtm